### PR TITLE
fixes #973 Better handling of renamed/transferred repositories #973

### LINKED ISSUE #973: Better handling of renamed/transferred repositories
If a repository is renamed or moved from one owner to another we currently don't do a good job of updating notifications `repository_full_name` or `owner_name` fields (unless the notifications are still in the user's github inbox, which is unlikely for users that have been on the site for over a month).

We do store the repository's remote id against each notification as well so we can potentially use that to find all notifications that need updating.

Probably the best place to put code to handle this is in an `after_save` callback on the Repository model that checks to see if the `full_name` has changed and then finds all notifications with `where(repository_id: repository.github_id)`  and update both `repository_full_name` and `owner_name` fields.

### PR DESCRIPTION
**Pull request by prashuchaudhary**
* Create a module to handle Notifications update if any of the repository is rename/transferred
* Added `activerecord-import` gem for doing bulk update
* Bulk update is done in batch of 1000
* Added Test case 

Fixes #973

### CODE REVIEW

**Comment by prashuchaudhary** [comment]
@andrew closed the previous one. Create new one. Sublime indentation was messed up.

File: lib/octobox/repository/update_notification_repository_name.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+      included do
+        after_update :upate_full_name_and_owner, if: -> { saved_change_to_full_name? }
+      end
+
+      # update full_name and owner_name of notifications if Repository full_name is updated
+      def upate_full_name_and_owner
+        notifications = Notification.where(repository_id: self.github_id)
+
+        return if notifications.blank?
```
[the remark this answers is not in the document]
Comment (prashuchaudhary) [reply lib/octobox/repository/update_notification_repository_name.rb]: sure.

**Comment by prashuchaudhary** [comment]
@andrew made the required changes

**Comment by andrew** [comment]
Great work as always!

**Pull request merged**